### ISSUE TITLE
Update  inc/html.class.php:829 

### ISSUE THREAD

**Issue by jdmarinz** [glpi-project/glpi#6671]
**Describe the bug**

If the user is in debug mode, this error appears systematically after a while.

           if ($with_session) {
               echo "<div id='debugsession$rand'>";
               self::printCleanArray($_SESSION, 0, true);
               echo "</div>";
            }

[2019-12-04 13:34:19] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  ...procedimientos/front/procedimiento.form.php:229 Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 46.61Mio)"}

**Page(s) URL**

When this error appears, it occurs on all pages.

**To reproduce**

This error happens at the time of using glpi, I cannot specify the event that triggers it
This error disappears if I restart my session
[2019-12-04 13:34:19] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  ...procedimientos/front/procedimiento.form.php:229 Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 46.61Mio)"}

**Expected behavior**

**Logs**

```
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Property access is not allowed yet
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Property access is not allowed yet
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Property access is not allowed yet
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Property access is not allowed yet
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:55:58] glpiphplog.ERROR: Toolbox::userErrorHandlerNormal() in /var/www/html/glpi941/inc/toolbox.class.php line 658
  *** PHP Warning(2): print_r(): Couldn't fetch mysqli
  Backtrace :
  :                                                  
  inc/html.class.php:3859                            print_r()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:3848                            Html::printCleanArray()
  inc/html.class.php:829                             Html::printCleanArray()
  inc/html.class.php:1671                            Html::displayDebugInfos()
  front/config.form.php:63                           Html::footer()
  {"user":"7@glpilab-des-sa1","mem_usage":"0.000\", 45.67Mio)"} 
[2019-12-04 13:56:09] glpiphplog.WARNING: Test logger  
```

**Screenshots**

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
[code]
 
GLPI 9.4.4 (/glpi941 => /var/www/html/glpi941)
Installation mode: TARBALL

```

**Additional context**

**Comment by trasher** [comment]
GLPI setup and version are missing.

Please try with up to date current bugfixes branch (or [latest nightly](https://forge.glpi-project.org/tarballs/))

**Comment by cedric-anne** [comment]
Hi,

This has been fixed in 9.5 branch by #6548.

Regards
